flow/raster_cache_util.cc

```diff
@@ -4,4 +4,67 @@
 
 #include "flutter/flow/raster_cache_util.h"
 
-namespace flutter {}
+namespace flutter {
+
+bool RasterCacheUtil::ComputeIntegralTransCTM(const SkMatrix& in,
```

Comment:
Do we need to make changes to the raster cache utils in this PR too?

Reply:
Why not?

I don't need to change it, but since the integral transform operation became really expensive due to the new work, the new form of these utility methods saves having to install an integral transform which saves the complicated work. With this change we only call `TransformReset` when we actually need to change the transform, which is a good thing.

BTW, none of the code paths that use this are executed for Impeller. This is to minimize the overhead on the older Skia pipeline.

Reply:
Because the large the scope of the PR, the more likely something is to break. I would prefer if you split that out into a second PR.

Reply:
I can work that in first and then rebase this PR on top of that.

Reply:
See https://github.com/flutter/engine/pull/51337

Reply:
https://github.com/flutter/engine/pull/51337 landed with these changes separated out and the latest commit here is rebased on top of it.